### ISSUE TITLE
Primer admits many empty names

### ISSUE THREAD

**Issue by dhess** [hackworthltd/primer#832]
## What happened?

While testing a frontend feature that implements a form-based type definition UI, I noticed that Primer admits typedefs whose `baseName` is empty. A truncated example log follows:

```
[INFO]      CreateTypeDef (Req (0b8708df-0070-40eb-8e0c-8f1701d15700,ExprTreeOpts {patternsUnder = True},GlobalName {qualifiedModule = ModuleName {unModuleName = "Main" :| []}, baseName = ""},[]))
[INFO]      Edit (Req (0b8708df-0070-40eb-8e0c-8f1701d15700,Edit [AddTypeDef (GlobalName {qualifiedModule = ModuleName {unModuleName = "Main" :| []}, baseName = ""}) (ASTTypeDef {astTypeDefParameters = [], astTypeDefConstructors = [], astTypeDefNameHints = []})]))
[INFO]      Edit (Resp (Right (Prog {progImports ....
```

## What did you expect to happen?

Primer should reject typedefs with an empty `baseName`.

## How can we reproduce it?

Trivial to reproduce and 100% repeatable. See the first commit of #833.

## How severe is the bug?

3 (can produce a program that is [pathologically] valid, but would be difficult to understand)

## Other notes

This issue does not only affect typedefs. It also affects:

* term definitions

Basically, anything that uses `Primer.Name.Name` is affected.
